types/k6/experimental/websockets.d.ts

```diff
@@ -37,13 +41,17 @@ export class WebSocket {
 
     /**
      * The Websocket constructor returns a newly created WebSocket object.
+     * https://k6.io/docs/javascript-api/k6-experimental/websockets/websocket/
      *
      * @param url - The URL to which to connect; this should be the URL to which the WebSocket server will respond.
+     * @param protocols - Either a single protocol string or an array of protocol strings. The param reserved for future use, currently ignored.
+     * @param params - Used for setting various WebSocket connection parameters such as headers, cookie jar, compression, etc.
      */
-    constructor(url: string);
+    constructor(url: string, protocols?: null, params?: Params | null);
```

Comment:
I'm not 100% sure about this. Still, I believe if you set the `?` on the type of the parameter, then the optional naturally translates to `Params | undefined | null` under the hood (Whereas setting it on the parameter name `params?: Params` does set it to `Params | undefined` under the hood, and would therefore justify the `| null` here).

```suggestion
    constructor(url: string, protocols?: null, params: Params?);
```

Reply:
Sorry, I'm not sure if I got the concern right :thinking:, but the current version says that the `params` [is optional](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#use-optional-parameters).

For example, consider the listed example of initiating the WebSocket:
```typescript
new WebSocket('wss://url');
new WebSocket('wss://url', 'somethingvalid');
new WebSocket('wss://test-api.k6.io/ws/crocochat/1', null, {
    tags: { user: 'zbt' }
});
```

All of them are valid, but if we use the definition:

```diff
+constructor(url: string, protocols: null, params: Params);
-constructor(url: string, protocols?: null, params?: Params);
```

The only third one:
```
new WebSocket('wss://test-api.k6.io/ws/crocochat/1', null, {
    tags: { user: 'zbt' }
});
```
will be valid, in other words, the user is expected to always provide all three arguments to the constructor.

Reply:
Actually, dismiss my document. I checked again, and my assumption was wrong. Sorry for the noise 🙇🏻 

For clarity: I was outlining that I thought that by writing the function definition as `constructor(url: string, protocol?: null, params: Params?)` it would automatically translate it to `constructor(url: string, protocol?: null, params?: Params | null)` under the hood. But after some experimentation, it seems I was wrong.